roles/tackle/tasks/main.yml

```diff
@@ -120,33 +119,87 @@
             state: present
             definition: "{{ lookup('template', 'secret-keycloak-sso.yml.j2') }}"
 
-    - name: "Setup Keycloak SSO Service"
-      k8s:
-        state: present
-        definition: "{{ lookup('template', 'service-keycloak-sso.yml.j2') }}"
+    - when:
+        - app_profile == "konveyor"
+      block:
+        - name: "Setup Keycloak SSO Service"
+          k8s:
+            state: present
+            definition: "{{ lookup('template', 'service-keycloak-sso.yml.j2') }}"
+
+        - name: "Setup Keycloak SSO Deployment"
+          k8s:
+            state: present
+            definition: "{{ lookup('template', 'deployment-keycloak-sso.yml.j2') }}"
 
-    - name: "Setup Keycloak SSO Deployment"
-      k8s:
-        state: present
-        definition: "{{ lookup('template', 'deployment-keycloak-sso.yml.j2') }}"
 - when:
     - feature_auth_required|bool
     - app_profile == "mta"
   block:
-    - name: "Check if RHSSO Keycloak CR exists already so we don't update it"
+    - name: "Check for existing RHSSO Keycloak CR"
       k8s_info:
         api_version: "{{ rhsso_api_version }}"
         kind: Keycloak
         namespace: "{{ app_namespace }}"
         label_selectors:
           - app = {{ rhsso_service_name }}
-      register: rhsso_keycloak_status
+      register: rhsso_keycloak
+
+    - name: "Delete old RHSSO Keycloak"
+      k8s:
+        state: absent
+        api_version: "{{ rhsso_api_version }}"
+        kind: Keycloak
+        name: "{{ rhsso_service_name }}"
+        namespace: "{{ app_namespace }}"
+      when:
+        - rhsso_keycloak.resources | length > 0
+        - '"keycloak-postgresql" in rhsso_keycloak.resources[0].status.secondaryResources.Deployment'
+
+    - name: "Get PostgreSQL Keycloak Secret"
+      k8s_info:
+        api_version: v1
+        kind: Secret
+        name: "{{ keycloak_database_secret_name }}"
+        namespace: "{{ app_namespace }}"
+      register: keycloak_database_secret
+
+    - name: "Get DNS operator CR"
+      k8s_info:
+        api_version: operator.openshift.io/v1
```

Comment:
How would this work with non-OpenShift: Keycloak on a vanilla k8s env?

Assume we need to at least wrap this with something to check for presence of this operator?

Reply:
I see now the logic is that "mta" differentiates what is invoked so "app_profile == "mta" block is for just the OpenShift case